src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -324,9 +324,8 @@ private void AddDirectoryWatch(WatchedDirectory parent, string directoryName)
                     // against the handle, so we'd deadlock if we relied on that approach.  Instead, we want to follow
                     // the approach of removing all watches when we're done, which means we also don't want to
                     // add any new watches once the count hits zero.
-                    if (parent == null || _wdToPathMap.Count > 0)
+                    if (_wdToPathMap.Count > 0)
```

Comment:
I've removed the `null` check because the method doesn't get called with `parent==null`, and the argument is also not nullable (`WatchedDirectory parent`).